pkg/bootstrap/bootstrap_config.go

```diff
@@ -311,7 +372,10 @@ func WriteBootstrap(config *meshconfig.ProxyConfig, node string, epoch int, pilo
 	// Support passing extra info from node environment as metadata
 	meta := getNodeMetaData(localEnv)
 
-	localityOverride := model.GetLocalityOrDefault("", meta)
+	localityOverride := ""
+	if locality, ok := meta[model.LocalityLabel].(string); ok {
```

Comment:
Have you tested this? Does the cast occur before checking ok or after? My concern is that the cast could happen on a nil value if it doesn't exist ... not sure if that would cause a panic.  I've just never seen this done before ... maybe go does the right thing here.

Reply:
moved the cast.